### PR TITLE
ASTPrinter: Turn on explicit `any` printing for everything and remove the option to disable it

### PR DESCRIPTION
**Pull request by AnthonyLatsis**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by AnthonyLatsis** [comment]
Heh, missed some after all.

**Comment by hborla** [comment]
>Do we want to flip the switch now? My impression (and it doesn’t have to be correct) has been that we want to flip it when Swift 6 comes around and any actually starts being required.

My intention was to only flip the switch for requiring it in source code in Swift 6. Diagnostics, Swift interfaces, and many other things already print the `any` keyword, even when it's not written in source. The only reason I added the ASTPrinter flag upfront was because there was a _lot_ to change, not because I deliberately wanted to suppress `any` printing before Swift 6. I also think that printing `any` in these places encourages folks to start writing `any` now.

>At least for code completion results, printing the any makes the results quite a bit more verbose.

I've always found it misleading that code completion displays existential parameters and generic parameters in the same way. But in any case, I think the desired behavior of code completion is independent from which language mode `any` becomes required in.

File: test/IDE/complete_operators.swift [hunk 1]
```diff
@@ -190,9 +190,9 @@ func testInfix12() {
 }
 // INFIX_12: Begin completions, 4 items
 // INFIX_12-DAG: Decl[AssociatedType]/CurrNominal:   .T; name=T
-// INFIX_12-DAG: Keyword[self]/CurrNominal:          .self[#P.Protocol#]; name=self
-// INFIX_12-DAG: Keyword/CurrNominal:                .Protocol[#P.Protocol#]; name=Protocol
-// INFIX_12-DAG: Keyword/CurrNominal:                .Type[#P.Type#]; name=Type
+// INFIX_12-DAG: Keyword[self]/CurrNominal:          .self[#(any P).Type#]; name=self
+// INFIX_12-DAG: Keyword/CurrNominal:                .Protocol[#(any P).Type#]; name=Protocol
```
Comment (ahoppen) [inline test/IDE/complete_operators.swift:194]: This should continue to be `P.Protocol`, right? I’m not sure what the type of `.self` and `.Type` should be, honestly, maybe for those `(any P).Type` is even correct.

File: test/SourceKit/CursorInfo/cursor_some_type.swift [hunk 2]
```diff
@@ -35,7 +35,8 @@ func test(value: S) {
 // PROTO_AFTER_SOME-NEXT: Proto
 // PROTO_AFTER_SOME-NEXT: s:4Test5ProtoP
 // PROTO_AFTER_SOME-NEXT: source.lang.swift
-// PROTO_AFTER_SOME-NEXT: Proto.Protocol
+// FIXME: InterfaceTypeRequest returns a meta constraint type for protocols.
```
Comment (ahoppen) [inline test/SourceKit/CursorInfo/cursor_some_type.swift:38]: I don’t quite understand that Fixme. What do you mean by it? I this something that broke with this change or something that has been broken before?

File: test/attr/attr_objc.swift [hunk 3]
```diff
@@ -1264,16 +1264,16 @@ class infer_instanceVar1 {
...
-// CHECK-LABEL: {{^}}  var var_ExistentialMetatype2: (PlainProtocol).Type
-// CHECK-LABEL: {{^}}  var var_ExistentialMetatype3: (PlainProtocol & Protocol_Class1).Type
-// CHECK-LABEL: {{^}}  var var_ExistentialMetatype4: (PlainProtocol & Protocol_ObjC1).Type
-// CHECK-LABEL: {{^}}  var var_ExistentialMetatype5: (Protocol_Class1).Type
-// CHECK-LABEL: {{^}}  var var_ExistentialMetatype6: (Protocol_Class1 & Protocol_Class2).Type
-// CHECK-LABEL: {{^}}  var var_ExistentialMetatype7: (Protocol_Class1 & Protocol_ObjC1).Type
-// CHECK-LABEL: @objc var var_ExistentialMetatype8: Protocol_ObjC1.Type
-// CHECK-LABEL: @objc var var_ExistentialMetatype9: (Protocol_ObjC1 & Protocol_ObjC2).Type
+// CHECK-LABEL: {{^}}  var var_ExistentialMetatype0: any Any.Type
+// CHECK-LABEL: {{^}}  var var_ExistentialMetatype1: any PlainProtocol.Type
```
Comment (ahoppen) [inline test/attr/attr_objc.swift:1268]: Just double-checking: Is `any PlainProtocol.Type` correct here or should this be `(any PlainProtocol).Type`? I don’t know what the right answer is.

File: test/IDE/complete_member_type.swift [hunk 4]
```diff
@@ -28,14 +28,14 @@ do {
   let _: Any.#^MEMBER_ANY_DOT^#
 
   // MEMBER_ANY_DOT-LABEL: Begin completions, 2 items
-  // MEMBER_ANY_DOT-NEXT: Keyword/None: Protocol[#Any.Protocol#]; name=Protocol
-  // MEMBER_ANY_DOT-NEXT: Keyword/None: Type[#Any.Type#]; name=Type
+  // MEMBER_ANY_DOT-NEXT: Keyword/None: Protocol[#(any Any).Type#]; name=Protocol
```
[the remark this answers is not in the document]
Comment (AnthonyLatsis) [reply test/IDE/complete_member_type.swift:31]: `Any` is short for `any Any` when used as a type vs. a constraint. We never print or require the full version, but you _can_ spell it in source. I don't remember if the proposal accentuates the existential `Any` metatype (I'm not near a laptop right now), but perhaps we might want to print it with `any` to guard against instances of muddling up `any Any.Type` (former `Any.Type`) with `(any Any).Type` (former `Any.Protocol)`.

File: test/attr/attr_objc.swift (hunk 3 above)
Comment (AnthonyLatsis) [reply test/attr/attr_objc.swift:1268]: The existential metatype is `any PlainProtocol.Type`, yes.

File: test/IDE/complete_operators.swift (hunk 1 above)
Comment (AnthonyLatsis) [reply test/IDE/complete_operators.swift:194]: `P.Protocol` is `(any P).Type` with `any` syntax, hence `P.self` is an expression of type `(any P).Type`. `.Type` currently always expects an existential metatype type as an outcome (for `[any] P` this is correct, but not for `(any P)`) . In Swift 6 mode, I think we should stop suggesting `[.]Protocol` and have two distinct `[.]Type` completions for a plain protocol/composition (see #65699).

File: test/SourceKit/CursorInfo/cursor_some_type.swift (hunk 2 above)
Comment (AnthonyLatsis) [reply test/SourceKit/CursorInfo/cursor_some_type.swift:38]: I guess I phrased it poorly, will fix. It was already this way and, I believe, broken (see [the complement FIXME](https://github.com/apple/swift/pull/65731/files/c1179155713e7504033fdf77b410885b64d5a828#diff-5a9439f8a77b136c97b430596bfc4263de54fbf7c0616656bbfa75ff2d10d145R2409)). The interface type of a protocol was always the protocol metatype aka `P.Protocol` (`(any P).Type` with `any` syntax), which is currently modeled as an `ExistentialType` wrapped in a `MetatypeType`. It seems we forgot to audit the request during the remodeling and still evaluate to a `ProtocolType` directly wrapped in a `MetatypeType`, which is not a sensible type in today’s Swift.

File: test/attr/attr_objc.swift (hunk 3 above)
Comment (ahoppen) [reply test/attr/attr_objc.swift:1268]: OK. Thanks for the clarification. 👍🏽

File: test/IDE/complete_operators.swift (hunk 1 above)
Comment (ahoppen) [reply test/IDE/complete_operators.swift:194]: I didn’t know you now spelled. `P.Protocol` as `(any P).Type`. In that case, makes sense to me.

File: test/IDE/complete_member_type.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (ahoppen) [reply test/IDE/complete_member_type.swift:31]: Uh, OK. With that explanation it somewhat makes sense. Let’s just leave it as-is and maybe stop suggesting `.Protocol` in Swift 6, which will just define the issue away.

**Review by ahoppen — APPROVED** [review]
LGTM for all the changes in `test/IDE`. I didn’t review the other changes.

File: test/IDE/complete_operators.swift (hunk 1 above)
Comment (AnthonyLatsis) [reply test/IDE/complete_operators.swift:194]: > `.Type` currently always expects an existential metatype type as an outcome (for `[any ]P` this is correct, but not for `(any P)`)

Reported in #65843.

Comment (ahoppen) [reply test/IDE/complete_operators.swift:194]: Thank you 🙏🏽

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test

**Comment by AnthonyLatsis** [comment]
@swift-ci Please test

**Review by xedin — APPROVED** [review]

**Comment by AnthonyLatsis** [comment]
@swift-ci Please test

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test macOS

**Pull request merged**